lighthouse-core/config/config-plugin.js

```diff
@@ -138,13 +139,17 @@ class ConfigPlugin {
     if (!i18n.isStringOrIcuMessage(manualDescription) && manualDescription !== undefined) {
       throw new Error(`${pluginName} has an invalid category manualDescription.`);
     }
+    if (!Array.isArray(supportedModes) && supportedModes !== undefined) {
```

Comment:
`Array.isArray()` unfortunately yields `Array<any>`, which is why it's allowed to be passed back as `Array<'navigation'|'timespan'|'snapshot'>` even though it might not be that.

Each item also needs to be checked, something like
```js
if (Array.isArray(supportedModes)) {
  const gatherModes = new Set(['navigation', 'timespan', 'snapshot']);
  if (!supportedModes
      .every(/** @return {m is LH.Gatherer.GatherMode} */ m => gatherModes.has(m))) {
    throw new Error(`${pluginName} has an invalid value in its category supportedModes.`);
  }
} else if (supportedModes !== undefined) {
  throw new Error(`${pluginName} has an invalid category supportedModes.`);
}
```
though hopefully that could be cleaned up a bit